aten/src/TH/generic/THTensorEvenMoreMath.cpp

```diff
@@ -221,8 +221,12 @@ void THTensor_(indexSelect)(THTensor *tensor, THTensor *src, int dim, THLongTens
 
   index = THLongTensor_newContiguous(index);
   index_data = THLongTensor_data(index);
+  bool src_is_contiguous = src->is_contiguous();
+  if (!src_is_contiguous) {
+    src = THTensor_(newContiguous)(src);
```

Comment:
Isn't this a performance regression in general for `index_select`?

Reply:
Also, I wonder if `index_select` is still useful speed-wise compared to advanced indexing? Might be good to do some benchmarks on it.

We could potentially just dispatch `index_select` to `index`

Reply:
@fmassa this is try to improve the DLRM performance on CPU path pulished at https://github.com/facebookresearch/dlrm. _embedding_bag_sparse_backward will call index_select. Maybe this can be used as one benchmark